Assembler.Base/BaseFrameHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Assembler.Core;
+using Assembler.Core.Entities;
+
+namespace Assembler.Base
+{
+    public abstract class BaseFrameHandler<TFrame, TMessage> : IFrameHandler<TFrame, TMessage>
+        where TFrame : BaseFrame
+        where TMessage : BaseMessageInAssembly
+    {
+        protected readonly ITimeBasedCache<TMessage> Cache;
+        protected readonly IFactory<TFrame, string> IdentifierFactory;
+        protected readonly IMessageEnricher<TFrame, TMessage> MessageEnricher;
+        protected readonly ICreator<TMessage> MessageInAssemblyCreator;
+
+        public event Action<TMessage> MessageAssemblyFinished;
+
+        public abstract void Handle(TFrame frame);
+
+        protected BaseFrameHandler(ITimeBasedCache<TMessage> cache, IFactory<TFrame, string> identifierFactory,
+            IMessageEnricher<TFrame, TMessage> messageEnricher, ICreator<TMessage> messageInAssemblyCreator)
+        {
+            Cache = cache;
+            IdentifierFactory = identifierFactory;
+            MessageEnricher = messageEnricher;
+            MessageInAssemblyCreator = messageInAssemblyCreator;
+        }
+
+        protected void ReleaseMessage(TMessage message)
```

Comment:
`ReleaseMessageInAssembly`, and udpate name of argument to messageInAssembly

Reply:
I don't think that `ReleaseMessageInAssembly` is the right name, I think the current name is better.

Reply:
So at least the name of the Interface should be MessageInAssemblyReleaser, something must be point that you release messageInAssembly, otherwise its not obvius

Reply:
f